docs/codeql/codeql-cli/about-codeql-packs.rst

```diff
@@ -56,49 +59,283 @@ The following properties are supported in ``qlpack.yml`` files.
+     - The path to a directory within the pack that contains database upgrade scripts, defined relative to the pack directory. Database upgrades are used internally to ensure a database created with a different version of the CodeQL CLI is compatible with the current version.
    * - ``authors``
-     - ``example@github.com``
+     - .. code-block:: yaml
+
+        authors: author1@github.com,author2@github.com
+
      - All packs
      - Metadata that will be displayed on the packaging search page in the packages section of the account that the CodeQL pack is published to.
-   * - ``licenses``
-     - ``(LGPL-2.1 AND MIT)``
+   * - ``license``
+     - .. code-block:: yaml
+
+        license: MIT
+
      - All packs
-     - Metadata that will be displayed on the packaging search page in the packages section of the account that the CodeQL pack is published to. For a list of allowed licenses, see `SPDX License List <https://spdx.org/licenses/>`__ in the SPDX Specification. 
+     - Metadata that will be displayed on the packaging search page in the packages section of the account that the CodeQL pack is published to. For a list of allowed licenses, see `SPDX License List <https://spdx.org/licenses/>`__ in the SPDX Specification.
    * - ``description``
-     - ``Human-readable description of the contents of the CodeQL pack.``
+     - .. code-block:: yaml
+
+        description: Human-readable description of the contents of the CodeQL pack.
+
      - All packs
      - Metadata that will be displayed on the packaging search page in the packages section of the account that the CodeQL pack is published to.
+   * - ``libraryPathDependencies``
+     - .. code-block:: yaml
+
+        libraryPathDependencies: codeql/javascript-all
+
+     - Optional, deprecated
+     - Use the ``dependencies`` property instead. The names of any CodeQL packs that this CodeQL pack depends on, as an array. This gives the pack access to any libraries, database schema, and query suites defined in the dependency.
+
+.. _about-codeql-pack-lock:
+
+About ``codeql-pack.lock.yml`` files
+------------------------------------
+
+``codeql-pack.lock.yml`` files store the versions of the resolved transitive dependencies of a CodeQL pack. This file is created by the ``codeql pack install`` if it does not already exist and should be added to your version control system. The ``dependencies`` section of the ``qlpack.yml`` file contains version ranges that are compatible with the pack. The ``codeql-pack.lock.yml`` file locks the versions to precise dependencies. This ensures that anyone running ``codeql pack install`` on this the pack will always retrieve the same versions of dependencies even if newer compatible versions exist. Only query packs can have ``codeql-pack.lock.yml`` files. This is because library packs are not compiled independently of query packs. Without being compiled, there is no need to specify precise versions of their dependencies.
```

Comment:
```suggestion
``codeql-pack.lock.yml`` files store the versions of the resolved transitive dependencies of a CodeQL pack. This file is created by the ``codeql pack install`` command if it does not already exist and should be added to your version control system. The ``dependencies`` section of the ``qlpack.yml`` file contains version ranges that are compatible with the pack. The ``codeql-pack.lock.yml`` file locks the versions to precise dependencies. This ensures that anyone running ``codeql pack install`` on this the pack will always retrieve the same versions of dependencies even if newer compatible versions exist.
```
Library packs do have lock files, at least in their source repo. This ensures deterministic pack resolution when running QL tests in library packs.

Reply:
Right. That's a subtlety that I forgot about. I will update this paragraph to make that clear.